### ISSUE TITLE
Clang-tidy: bugprone-use-after-move: false negative

### ISSUE THREAD

**Issue by JVApen** [llvm/llvm-project#59612]
Reproduction:
````cpp
// https://compiler-explorer.com/z/8asKxxeGh
// clang++ -std=c++20  -stdlib=libc++  -Weverything -Wno-c++98-compat -Wno-missing-prototypes t.cpp
// clang-tidy -checks=bugprone-use-after-move
#include <utility>

struct C {};

struct S {
    auto g(C c) -> void;
};

auto f(C &c) -> S;

int main(int, char**) {
    auto c = C{};
    f(c).g(std::move(c));
}

````

Expected:
````console
[<source>:13:7: warning: 'c' used after it was moved [bugprone-use-after-move]]
    f(c).g(std::move(c));
      ^
[<source>:13:12: note: move occurred here]
    f(c).g(std::move(c));
           ^
[<source>:13:7: note: the use and move are unsequenced, i.e. there is no guarantee about the order in which they are evaluated]
    f(c).g(std::move(c));
      ^
1 warning generated.
````

When the argument of `f` is changed from `C &c` to either `const C &c` or `C c`, the expected warning shows up. However, when the class is an (in/)out argument, this does not show. I don't believe there is a difference in the sequence of execution for these cases, so the warning should happen in all cases.

**Comment by JVApen** [comment]
@dotnwat You might be right about that. I forgot about that change. Based on that, it implies this isn't a false negative but a false positive. 

I've checked the proposals:
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0145r3.pdf

> In summary, the following expressions are evaluated in the order a, then b, then c, then d:
> 1. a.b
> 2. a->b
> ...

So this indeed seems like 2 false positives for C++17 and above, and 1 false negative for C++14 and below.

**Comment by PiotrZSL** [comment]
https://reviews.llvm.org/D145581
Reviewers wanted...
